File: 2 semestre/Técnica de Desenvolvimento de Algoritmos/Listas de Exercicios/Lista 3/ex10.c
#include <stdio.h>

main()
{
    printf("Numero de dias entre duas datas (1950-1996)\n\n");

    //vars
    int d1, m1, a1, d2, m2, a2, aux1, aux2, aux3, aI, totalD1=0, totalD2=0, totalDias=0;

    //Entrada
    printf("Insira o ano da 1o data: ");
    scanf("%d", &a1);

    printf("Insira o mes da 1o data: ");
    scanf("%d", &m1);

    printf("Insira o dia da 1o data: ");
    scanf("%d", &d1);

    system("cls");

    printf("Insira o ano da 2o data: ");
    scanf("%d",&a2);

    printf("Insira o mes da 2o data: ");
    scanf("%d", &m2);

    printf("Insira o dia da 2o data: ");
    scanf("%i", &d2);

    //Processamento

    //Trocando as datas, se necessário
    if(a1 > a2)
    {
        aux1 = a2;
        aux2 = m2;
        aux3 = d2;
        a2 = a1;
        m2 = m1;
        d2 = d1;
        a1 = aux1;
        m1 = aux2;
        d1 = aux3;
    }
    aI = a1;

    if(a1!=a2)
    {
        if(m1 != 1)
            m1--;
        //Quantidade de dias para terminar o ano da data inicial
        totalD1 = 365 - (m1*30 + d1);
        //Quantidade de dias passados do inicio do ano da data final
        totalD2 = ((m2-1)*30 + d2);
        a1++;
    }
    else
    {
        if(m2-1-m1 == 0)
            m2++;
        totalD2 = (m2-1-m1)*30 + (d2-d1);
    }

    totalDias = totalD1 + totalD2;

    while (a1 < a2)
    {
        if((1952-a1)%4 == 0 || a2==1952)
        {
            totalDias += 366;
        }
        else
        {
            totalDias += 365;
        }

        a1++;

    }
    //Retornando a1 ao seu valor principal
    a1 = aI;


    //Saída
    printf("\nData Inicial: %i/%i/%i\n",d1, m1, aI);
    printf("Data Final: %d/%d/%d\n", d2, m2, a2);
    printf("Total de dias: %d", totalDias);



}

File: 2 semestre/Laboratório de Desenvolvimento de Algoritmos/Lista de Exercícios 3/src/Ex10.java
//Importando Scanner
import java.util.Scanner;

public class Ex10{
	
	public static void main(String[] args){
		
		//Instanciando Scanner
		Scanner input = new Scanner(System.in);
		
		System.out.println("Dias entre duas datas\n\n");
		
		//vars
		int a1, m1, d1, a2, m2, d2, dF, aI, totalDias=0, auxAno, auxMes, auxDia;
		
		//Entrada
		System.out.print("Digite o ano inicial (>1950): ");
		a1 = input.nextInt();
		
		System.out.print("Digite o mês: ");
		m1 = input.nextInt();
		
		System.out.print("Digite o dia: ");
		d1 = input.nextInt();
		
		System.out.print("\n\nDigite o ano final(>1996): ");
		a2 = input.nextInt();
		
		System.out.print("Digite o mês: ");
		m2 = input.nextInt();
		
		System.out.print("Digite o dia: ");
		d2 = input.nextInt();
		
		//Processamento
		
		//Trocando as datas, caso estejam invertidas
		if ( (a1>a2) || (a1==a2 && m1>m2) ||  (a1==a2 && m1==m2 && d1>d2))
		{
			auxAno = a1;
			auxMes = m1;
			auxDia = d1;
			a1 = a2;
			m1 = m2;
			d1 = d2;
			a2 = auxAno;
			m2 = auxMes;
			d2 = auxDia;
		}
		
		//totalDias = Quantidade de dias já decorridas do ano atual + quantidade de dias decorridas do ano final
		totalDias = (365 - ( (m1-1)*30 + d1 )) + ( (m2-1)*30 + d2 );
		
		//Preservado o valor do ano inicial, para ser usado no final
		aI = a1 + 1;
		
		while( aI < a2  )
		{
			//Se o ano for bissexto
			if( (aI-1952)%4 == 0 )
				totalDias += 366;
			else
				totalDias += 365;	
		}
		
		//Saída
		System.out.printf("\nAno inicial: %d/%d/%d\n", d1,m1,a1);
		System.out.printf("Ano final: %d/%d/%d\n", d2,m2,a2);
		System.out.println("Total de dias: " + totalDias);
		
		
		
		
		
	}
}